utils/wsutil/ws.go

```diff
@@ -157,50 +157,37 @@ func (ws *Websocket) SendCtx(ctx context.Context, b []byte) error {
 // Close closes the websocket connection. It assumes that the Websocket is
 // closed even when it returns an error. If the Websocket was already closed
 // before, ErrWebsocketClosed will be returned.
-func (ws *Websocket) Close() error {
-	WSDebug("Conn: Acquiring mutex lock to close...")
-
-	ws.mutex.Lock()
-	defer ws.mutex.Unlock()
-
-	WSDebug("Conn: Write mutex acquired; closing.")
+func (ws *Websocket) Close() error { return ws.close(false) }
 
-	return ws.close()
-}
+func (ws *Websocket) CloseGracefully() error { return ws.close(true) }
 
-func (ws *Websocket) CloseGracefully() error {
-	WSDebug("Conn: Acquiring mutex lock to close gracefully...")
+// close closes the Websocket without acquiring the mutex. Refer to Close for
+// more information.
+func (ws *Websocket) close(graceful bool) error {
+	WSDebug("Conn: Acquiring mutex lock to close...")
 
 	ws.mutex.Lock()
 	defer ws.mutex.Unlock()
 
 	WSDebug("Conn: Write mutex acquired")
 
-	if gc, ok := ws.conn.(GracefulCloser); ok {
-		if ws.closed {
-			WSDebug("Conn: Websocket is already closed.")
-			return ErrWebsocketClosed
-		}
-
-		WSDebug("Conn: closing gracefully")
-
-		ws.closed = true
-		return gc.CloseGracefully()
-	} else {
-		WSDebug("Conn: The Websocket's Connection does not provide graceful closure. Closing normally instead.")
-		return ws.close()
-	}
-}
-
-// close closes the Websocket without acquiring the mutex. Refer to Close for
-// more information.
-func (ws *Websocket) close() error {
 	if ws.closed {
 		WSDebug("Conn: Websocket is already closed.")
 		return ErrWebsocketClosed
 	}
 
-	err := ws.conn.Close()
 	ws.closed = true
-	return err
+
+	if graceful {
+		if gc, ok := ws.conn.(GracefulCloser); ok {
+			WSDebug("Conn: Closing gracefully")
+			return gc.CloseGracefully()
+		}
+
+		WSDebug("Conn: The Websocket's Connection does not support graceful closure. Closing normally instead.")
+		return ws.conn.Close()
```

Comment:
This like can just not be here; execution will fall out into the `Close()` after it. The redundant debug logging is fine.

Reply:
I don't understand, please elaborate.

Reply:
Like this:

```go
	if graceful {
		if gc, ok := ws.conn.(GracefulCloser); ok {
			WSDebug("Conn: closing gracefully")
			return gc.CloseGracefully()
		}

		WSDebug("Conn: graceful closure unsupported; closing normally instead.")
	}

	WSDebug("Conn: closing")
	return ws.conn.Close()
```

Reply:
Sure, I just wanted to prevent double logging, but I guess `WSDebug` is verbose enough as for this to not matter. Again, I will do this tomorrow